src/main/java/seedu/address/model/Model.java

```diff
@@ -99,6 +103,25 @@
      */
     void updateFilteredPersonList(Predicate<Person> predicate);
 
+    // ====== Contact ======
+
+    /**
+     * Returns true if the contact exists in the list.
+     */
+    boolean hasContact(Contact contact);
+
+    /**
+     * Deletes the given contact.
+     * The contact must exist in the list.
+     */
+    void deleteContact(Contact contact);
+
+    /**
+     * Adds the given contact.
+     * {@code contact} must not exist in the list.
+     */
+    void addContact(Contact contact);
+
```

Comment:
Please add the methods setContact(Contact target, Contact editedContact), getFilteredContactList(), updateFilteredContactList(Predicate<Contact> predicate), and implement these in `ModelManager`